scripts/gen_app_partitions.py

```diff
@@ -22,15 +22,22 @@
 linker_start_seq = """
 	SECTION_PROLOGUE(_APP_SMEM_SECTION_NAME, (OPTIONAL),)
 	{
+#ifdef CONFIG_ARC
```

Comment:
This fixes the alignment for ARC, but an architecture neutral solution in the generic script would be better. I'm not an ARM expert, but the fixed value of 4 below might be wrong as well since it doesn't match e.g. the ARMv8-M 32 byte MPU granularity.

Reply:
Actually this is not a problem. This generated code sits just after an alignment instruction in the linker. The alignment is done by app_data_alignment.ld. 
Which will align to what ever boundary needed for the current device. 
So I feel this should work. 
@ioannisg @agross-korg Please comment if I am mistaken here.

Reply:
Adding @agross-linaro